stdlib/public/core/StringUTF16View.swift

```diff
@@ -201,6 +201,17 @@ extension String.UTF16View: BidirectionalCollection {
       return _foreignIndex(i, offsetBy: n)
     }
 
+    if n.magnitude <= _StringBreadcrumbs.breadcrumbStride {
+      // Do not use breadcrumbs if directly computing the result is expected to
+      // be cheaper.
+      if _guts.isASCII {
```

Comment:
Should the isASCII check be hoisted? If we have ASCII, we don't care how big `n` is, do we?

Reply:
I was reluctant to move the similar check out of the `_nativeGetOffset`/`_nativeGetIndex` functions (to avoid having to audit all their callers), so rather than doing that I added a similar path here.

A simpler way of doing it would be to just add `!_guts.isASCII` to the outer condition. 🤔

```swift
    if n.magnitude <= _StringBreadcrumbs.breadcrumbStride, !_guts.isASCII {
      // Do not use breadcrumbs if directly computing the result is expected to
      // be cheaper.
      return _index(i, offsetBy: n)._knownUTF8
    }
```

Reply:
Yeah, that's what I meant by "hoist" =)